util/tracing/detect/detect.go

```diff
@@ -29,8 +30,6 @@ type detector struct {
 var ServiceName string
 var Recorder *TraceRecorder
 
-var Resource *resource.Resource
```

Comment:
Looks like this is reverting https://github.com/moby/buildkit/pull/4496, which will likely break moby, or is there another approach that moby should be taking after this patch?

Reply:
The `OverrideResource` method should be used instead to manually set the resource during initialization.

My hope is that a future refactor will remove the need for this function. I'm working on a separate refactor that removes some of the global state so the tracers and metrics exporters can be initialized with a user-provided resource, but that's not ready yet.

Reply:
Ah! Gotcha. Missed that one (blame phones and tiny screens) 😅